core/client/hdfs/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -225,6 +265,80 @@ public void resetContext() throws Exception {
+        fsConf.get(PropertyKey.ZOOKEEPER_ADDRESS.toString()));
+    verify(mMockFileSystemContext).reset(alluxio.Configuration.global());
+  }
+
+  @Test
+  public void resetContextUsingMultipleZkUriWithSemicolons() throws Exception {
+    URI uri = URI.create(Constants.HEADER + "zk@host1:port1;host2:port2;host3:port3/tmp/path.txt");
+    org.apache.hadoop.fs.FileSystem fs = org.apache.hadoop.fs.FileSystem.get(uri, getConf());
+    Configuration fsConf = fs.getConf();
+    assertTrue(fsConf.getBoolean(PropertyKey.ZOOKEEPER_ENABLED.toString(), false));
+    assertEquals("host1:port1,host2:port2,host3:port3",
+        fsConf.get(PropertyKey.ZOOKEEPER_ADDRESS.toString()));
+    verify(mMockFileSystemContext).reset(alluxio.Configuration.global());
+  }
+
+  @Test
+  public void resetContextFromSingleZkUriToAnotherSingleZkUri() throws Exception {
+    URI uri = URI.create(Constants.HEADER + "zk@host:port/tmp/path.txt");
+    Configuration fsConf = org.apache.hadoop.fs.FileSystem.get(uri, getConf()).getConf();
+    assertEquals("host:port", fsConf.get(PropertyKey.ZOOKEEPER_ADDRESS.toString()));
+
+    URI otherUri = URI.create(Constants.HEADER + "zk@otherHost:otherPort/tmp/path.txt");
+    Configuration otherFsConf = org.apache.hadoop.fs.FileSystem.get(otherUri, getConf()).getConf();
+    assertEquals("otherHost:otherPort", otherFsConf.get(PropertyKey.ZOOKEEPER_ADDRESS.toString()));
+    verify(mMockFileSystemContext, times(2)).reset(alluxio.Configuration.global());
+  }
+
+  @Test
+  public void resetContextFromSingleZkUriToMultipleZkUri() throws Exception {
+    URI uri = URI.create(Constants.HEADER + "zk@host:port/tmp/path.txt");
+    Configuration fsConf = org.apache.hadoop.fs.FileSystem.get(uri, getConf()).getConf();
+    assertEquals("host:port", fsConf.get(PropertyKey.ZOOKEEPER_ADDRESS.toString()));
+
+    URI otherUri = URI.create(Constants.HEADER
+        + "zk@host1:port1,host2:port2,host3:port3/tmp/path.txt");
+    Configuration otherFsConf = org.apache.hadoop.fs.FileSystem.get(otherUri, getConf()).getConf();
+    assertEquals("host1:port1,host2:port2,host3:port3",
+        otherFsConf.get(PropertyKey.ZOOKEEPER_ADDRESS.toString()));
+    verify(mMockFileSystemContext, times(2)).reset(alluxio.Configuration.global());
+  }
+
+  @Test
+  public void resetContextFromZkUriToNonZkUri() throws Exception {
+    URI uri = URI.create(Constants.HEADER + "zk@zkHost:zkPort/tmp/path.txt");
+    Configuration fsConf = org.apache.hadoop.fs.FileSystem.get(uri, getConf()).getConf();
+    assertTrue(fsConf.getBoolean(PropertyKey.ZOOKEEPER_ENABLED.toString(), false));
+    assertEquals("zkHost:zkPort", fsConf.get(PropertyKey.ZOOKEEPER_ADDRESS.toString()));
+
+    URI otherUri = URI.create(Constants.HEADER + "alluxioHost:alluxioPort/tmp/path.txt");
+    Configuration otherFsConf = org.apache.hadoop.fs.FileSystem.get(otherUri, getConf()).getConf();
+    assertFalse(otherFsConf.getBoolean(PropertyKey.ZOOKEEPER_ENABLED.toString(), false));
```

Comment:
should the default be `true` here? Otherwise we might not be checking what we think we're checking

Reply:
Is `Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)` false? We don't actually use the hadoop configuration after initialization, so the configuration returned by `getConf()` isn't really relevant